statsd/end_to_end_udp_test.go

```diff
@@ -106,6 +112,70 @@ func TestKnownEnvTagsEmptyString(t *testing.T) {
 	ts.sendAllAndAssert(t, client)
 }
 
+func TestPrefixedOriginWithEntityID(t *testing.T) {
+	entityIDEnvName := "DD_ENTITY_ID"
+
+	defer func() { os.Unsetenv(entityIDEnvName) }()
+
+	os.Setenv(entityIDEnvName, "pod-uid")
+
+	containerID = func() string { return "fake-container-id" }
+	defer func() { containerID = dummyContainerID }()
+
+	expectedTags := []string{"dd.internal.entity_id:pod-uid"}
+	ts, client := newClientAndTestServer(t,
+		"udp",
+		"localhost:8765",
+		expectedTags,
+	)
+
+	sort.Strings(client.tags)
+	assert.Equal(t, expectedTags, client.tags)
+	ts.sendAllAndAssert(t, client)
+}
+
+func TestPrefixedOriginWithoutEntityID(t *testing.T) {
+	entityIDEnvName := "DD_ENTITY_ID"
+	os.Unsetenv(entityIDEnvName)
+
+	containerID = func() string { return "fake-container-id" }
+	defer func() { containerID = dummyContainerID }()
+
+	expectedTags := []string{"dd.internal.prefixed_origin:container_id://fake-container-id"}
+	ts, client := newClientAndTestServer(t,
+		"udp",
+		"localhost:8765",
+		expectedTags,
+	)
+
+	sort.Strings(client.tags)
+	assert.Equal(t, expectedTags, client.tags)
+	ts.sendAllAndAssert(t, client)
+}
+
+func TestPrefixedOriginDisabled(t *testing.T) {
```

Comment:
Coudl we add a check where both `DD_ENTITY_ID` and `DD_UDP_ORIGIN_DISABLED` are set ?